src/morphological_tagger/MorphDictionary.h

```diff
@@ -0,0 +1,121 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with TurboParser 2.3.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef MORPHDICTIONARY_H_
+#define MORPHDICTIONARY_H_
+
+#include "SequenceDictionary.h"
+#include "TokenDictionary.h"
+#include "MorphReader.h"
+
+class MorphDictionary : public SequenceDictionary {
+public:
+  MorphDictionary() {}
+  MorphDictionary(Pipe* pipe) : SequenceDictionary(pipe) {}
+  virtual ~MorphDictionary() {}
+
+  void Clear() {
+    SequenceDictionary::Clear();
+    cpostag_morphtags_.clear();
+  }
+
+  void Save(FILE *fs) {
+    SequenceDictionary::Save(fs);
+    bool success;
+    int length = unknown_cpostag_morphtags_.size();
+    success = WriteInteger(fs, length);
+    CHECK(success);
+    for (int j = 0; j < unknown_cpostag_morphtags_.size(); ++j) {
+      int tag = unknown_cpostag_morphtags_[j];
+      success = WriteInteger(fs, tag);
+      CHECK(success);
+    }
+
+    length = cpostag_morphtags_.size();
+    success = WriteInteger(fs, length);
+    CHECK(success);
+    for (int i = 0; i < cpostag_morphtags_.size(); ++i) {
+      length = cpostag_morphtags_[i].size();
+      success = WriteInteger(fs, length);
+      CHECK(success);
+      for (int j = 0; j < cpostag_morphtags_[i].size(); ++j) {
+        int tag = cpostag_morphtags_[i][j];
+        success = WriteInteger(fs, tag);
+        CHECK(success);
+      }
+    }
+  }
+
+  void Load(FILE *fs) {
+    SequenceDictionary::Load(fs);
+    bool success;
+    int length;
+    success = ReadInteger(fs, &length);
+    CHECK(success);
+    unknown_cpostag_morphtags_.resize(length);
+    for (int j = 0; j < unknown_cpostag_morphtags_.size(); ++j) {
+      int tag;
+      success = ReadInteger(fs, &tag);
+      CHECK(success);
+      unknown_cpostag_morphtags_[j] = tag;
+    }
+    success = ReadInteger(fs, &length);
+    CHECK(success);
+    cpostag_morphtags_.resize(length);
+    for (int i = 0; i < cpostag_morphtags_.size(); ++i) {
+      success = ReadInteger(fs, &length);
+      CHECK(success);
+      cpostag_morphtags_[i].resize(length);
+      for (int j = 0; j < cpostag_morphtags_[i].size(); ++j) {
+        int tag;
+        success = ReadInteger(fs, &tag);
+        CHECK(success);
+        cpostag_morphtags_[i][j] = tag;
+      }
+    }
+  }
+
+  void CreateTagDictionary(MorphReader *reader);
+
+  const vector<int> &GetCpostagMorphtags(int cpostag) {
+    // return cpostag_morphtags_[cpostag];
+    // TODO: Not sure is this should be done here...
+    // It may be cleaner to return an empty vector here and 
+    // fill it with the unknown tags elsewhere.
+    if (!cpostag_morphtags_[cpostag].empty()) {
+      return cpostag_morphtags_[cpostag];
+    }
+    else {
+      return unknown_cpostag_morphtags_;
+    }
+  }
+
+protected:
+  vector<vector<int> > cpostag_morphtags_;
```

Comment:
Use std::vector, std::string instead of vector, string. In the future I will remove all the "using namespace std" to comply with this guideline: https://google.github.io/styleguide/cppguide.html#Namespaces.